airflow/dag_processing/manager.py

```diff
@@ -398,18 +406,20 @@ def _fetch_callbacks(
 
     def _add_callback_to_queue(self, request: CallbackRequest):
         self.log.debug("Queuing %s CallbackRequest: %s", type(request).__name__, request)
-        self.log.warning("Callbacks are not implemented yet!")
-        # TODO: AIP-66 make callbacks bundle aware
-        return
-        self._callback_to_execute[request.full_filepath].append(request)
-        if request.full_filepath in self._file_path_queue:
-            # Remove file paths matching request.full_filepath from self._file_path_queue
-            # Since we are already going to use that filepath to run callback,
-            # there is no need to have same file path again in the queue
+        bundle = self._bundles_manager.get_bundle(name=request.bundle_name, version=request.bundle_version)
+        dag_absolute_path = os.fspath(Path(bundle.path, request.filepath))
+        file_info = DagFileInfo(
+            path=dag_absolute_path, bundle_name=request.bundle_name, bundle_version=request.bundle_version
+        )
+        self._callback_to_execute[file_info].append(request)
+        if file_info in self._file_path_queue:
+            # Remove DagFileInfo matching DagFileInfo from self._file_path_queue
+            # Since we are already going to use that DagFileInfo to run callback,
+            # there is no need to have same DagFileInfo again in the queue
             self._file_path_queue = deque(
-                file_path for file_path in self._file_path_queue if file_path != request.full_filepath
+                file_path for file_path in self._file_path_queue if file_path != file_info
             )
```

Comment:
We don't need to do this anymore. Leave the existing entry, if it's in there, in there. Just add the versioned one to run the callback.